examples/audio/lhotse_example.py

```diff
@@ -0,0 +1,235 @@
+                    break
+                else:
+                    # No. We'll warn the user that the constrains might be too tight,
+                    # and return the cut anyway.
+                    warnings.warn(
+                        "The first cut drawn in batch collection violates "
+                        "the max_frames, max_cuts, or max_duration constraints - "
+                        "we'll return it anyway. "
+                        "Consider increasing max_frames/max_cuts/max_duration."
+                    )
+                    cuts.append(next_cut)
+
+        self.diagnostics.keep(cuts)
+        return CutSet.from_cuts(cuts)
+
+
+class IODataPipe(dp.IterDataPipe):
+    def __init__(self, datapipe):
+        self.datapipe = datapipe
+
+    def __iter__(self):
+        for cut_idx, batch_idx, batch_size, cut in self.datapipe:
+            yield cut_idx, batch_idx, batch_size, cut.load_audio(), cut
+
+
+class RecombineBatchAfterIO(dp.IterDataPipe):
+    def __init__(self, datapipe):
+        self.datapipe = datapipe
+
+    def __iter__(self):
+        # This is a buffer that will be re-combining batches.
+        # We might possibly get cuts from the same batch out-of-order.
+        batches: Dict[int, List[Cut]] = defaultdict(list)
+        for cut_idx, batch_idx, batch_size, audio, cut in self.datapipe:
+            batches[batch_idx].append((audio, cut))
+            if len(batches[batch_idx]) == batch_size:
+                audios, cuts = zip(*batches[batch_idx])
+                yield audios, CutSet.from_cuts(cuts)
+                # Free up the buffer dict after using the batch.
+                del batches[batch_idx]
+
+
+class UnbatchForIO(dp.IterDataPipe):
+    def __init__(self, datapipe):
+        self.datapipe = datapipe
+
+    def __iter__(self):
+        for batch_idx, cuts_batch in enumerate(self.datapipe):
+            batch_size = len(cuts_batch)
+            for cut_idx, cut in enumerate(cuts_batch):
+                yield cut_idx, batch_idx, batch_size, cut
+
+
+def classifier_fn(value, num_jobs):
+    return value[0] % num_jobs
+
+
+def build_pipeline(path: str, num_jobs: int = 2):
+
+    # Open cuts manifest for lazy reading
+    datapipe = CutsReader(path)
+
+    # Custom Sampling
+    datapipe = DurationBatcher(datapipe, max_duration=100)
+
+    # Unbatch the batched to run I/O
+    datapipe = UnbatchForIO(datapipe)  # Yield (batch index, dataclass)
+
+    # The following Multiprocessing API may change, and multiprocessing should be added and handled by DataLoader
+    # demux and mux would be called by DataLoader. But, I want to explicitly call them to illustrate the functionality
+
+    # Routed prior datapipe to ``num_jobs`` datapipes in round-robin manner
+    datapipes = datapipe.demux(
+        num_jobs, classifier_fn=partial(classifier_fn, num_jobs=num_jobs)
+    )
+    for i in range(len(datapipes)):
+        datapipes[i] = IODataPipe(datapipes[i])
```

Comment:
Thank you so much. This is a great example to show the functionality to distribute extensive operation to multiprocessing.